File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/subject/SubjectSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.subject;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface SubjectSubcomponent
    extends AndroidInjector<SubjectFragment> {

  void inject(SubjectPresenterImpl subjectPresenter);

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<SubjectFragment> {
    @Override public abstract SubjectSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/subject/SubjectSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.subject

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface SubjectSubcomponent : AndroidInjector<SubjectFragment> {

  fun inject(subjectViewModel: SubjectViewModel)

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<SubjectFragment>() {
    abstract override fun build(): SubjectSubcomponent
  }
}
